### PR TITLE
Update the changelog when creating an RC

### PR DESCRIPTION
**Pull request by Gudahtt**
The changelog will now be automatically updated when a release branch is created. A new release header along with changelog entries for any new commits will be added.

Note that this changelog will still need to be manually cleaned up, but it's one less manual step at least.

The old Bash script for adding a new release header to the changelog has been removed, as that functionality is now built into the changelog update script.

A new script has been added to commit any changes made to the manifest and changelog. This step used to happen at the end of the bump manifest version script, but now the changelog update relies upon the manifest version bump happening first, so it needed to be re-ordered. The changes should only be committed on the first run of the branch, as it's contingent upon the manifest changing (due to the version bump). Further changelog updates won't trigger new automatic commits.

Relates to #10752

Manual testing steps:  
  - Create a new release branch
  - See that it updates the changelog automatically

### CODE REVIEW

**Comment by Gudahtt** [comment]
This depends upon #10794

File: .circleci/scripts/release-commit-verison-bump.sh [hunk 1]
```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env bash
```
Comment (rekmarks) [inline .circleci/scripts/release-commit-verison-bump.sh:1]: Typo in file name: `verison` -> `version`.

It appears to be correctly spelled in the CircleCI config, but we should review all references.

**Review by rekmarks — CHANGES_REQUESTED** [review]
Looks good except for a typo.

File: .circleci/scripts/release-commit-verison-bump.sh (hunk 1 above)
Comment (Gudahtt) [reply .circleci/scripts/release-commit-verison-bump.sh]: Oops :facepalm: Good catch! Fixed.

**Review by rekmarks — DISMISSED** [review]
LGTM!

**Comment by Gudahtt** [comment]
I've just created the branch `Version-v9.3.1` to test this, just to double check that it all works :eyes:

**Comment by rekmarks** [comment]
ffffuuuuuuu
```
@lavamoat/allow-scripts automatically updating configuration

removing unneeded configuration for packages:
- fsevents
Done in 2.65s.
allow-scripts configuration requires updates
```

File: .circleci/scripts/release-commit-version-bump.sh [hunk 2]
```diff
@@ -18,7 +18,7 @@ fi
 
 printf '%s\n' 'Commit the manifest version and changelog if the manifest has changed'
 
-if ! git diff --quiet CHANGELOG.md app/manifest/_base.json;
+if git diff --quiet app/manifest/_base.json;
```
Comment (rekmarks) [inline .circleci/scripts/release-commit-version-bump.sh:21]: 😓

**Comment by Gudahtt** [comment]
It worked! https://github.com/MetaMask/metamask-extension/commit/d6b5abdc68eece5ed281600900698bc694633767

... after fixing a couple more mistakes :sweat_smile: . I put each bug fix in a separate commit - check out the most recent two since your approval to see what they were.

**Comment by Gudahtt** [comment]
That status check shouldn't affect this PR thankfully, because this isn't a release branch. It was only triggered because this commit was used in a release branch. GitHub applies these status checks by commit it seems, not by branch.

**Review by rekmarks — APPROVED** [review]
Alright, now it looks good!

**Pull request merged**